### PR TITLE
fix: handle multi-allelic gnomAD AF values in patch.py

### LINKED ISSUE #58: bug: float conversion fails on multi-allelic gnomAD AF values in patch.py
## Bug

`genechat init --fast` (and non-fast) crashes during the gnomAD population frequencies annotation step:

```
Unexpected error: could not convert string to float: '0.25,.'
```

## Root Cause

In `src/genechat/patch.py` lines 339-340:

```python
float(af) if af else None,
float(af_grpmax) if af_grpmax else None,
```

`_extract_info_field()` returns the raw VCF INFO string for multi-allelic sites. For a triallelic site, gnomAD's `Number=A` AF field contains comma-separated values like `0.25,.` (one allele with freq 0.25, one missing). `float("0.25,.")` raises `ValueError`.

## How to Reproduce

Run `genechat init <vcf> --label <name> --gnomad --dbsnp --fast` on the GIAB NA12878 VCF. The crash occurs during the gnomAD population frequencies substep after all downloads complete (~3 hours into the pipeline on a Fly.io performance-2x).

## Why It Wasn't Caught Earlier

The pre-patch.db architecture (before commit `6b99da2`) used `bcftools annotate -Oz -o annotated.vcf.gz` and pysam for reading, which correctly parsed `Number=A` fields as Python tuples. The new patch.db pipeline manually parses the raw VCF text stream where multi-allelic values are raw comma-separated strings.

The test in `test_patch.py` (line 155, `test_update_gnomad`) only uses single-allele AF values.

## Suggested Fix

A small helper that splits on comma and handles missing values:

```python
def _parse_af(raw: str | None) -> float | None:
    if not raw:
        return None
    for part in raw.split(","):
        part = part.strip()
        if part and part != ".":
            try:
                return float(part)
            except ValueError:
                continue
    return None
```

Then replace `float(af) if af else None` with `_parse_af(af)` on lines 339-340.

Note: `gwas.py` has a `_safe_float` pattern (line 71) but it doesn't handle comma-separated values.

## Impact

Blocks all annotation pipelines that encounter multi-allelic sites with gnomAD AF annotations — which is essentially every real genome. The GIAB NA12878 VCF hits this reliably.

## Context

Discovered during genechat-mcp-remote-demo annotation on Fly.io ([ADR-0003 workflow](https://github.com/natecostello/genechat-mcp-remote-demo/blob/main/docs/plans/deploy-and-annotate.md)). The annotation ran for ~3 hours, completed all downloads (gnomAD ~180 GB, dbSNP ~28 GB, ClinVar, SnpEff) and the dbSNP contig rename (~2 hrs), then crashed at the very last annotation substep.

### PR DESCRIPTION
**Pull request by natecostello**
## Summary
- Multi-allelic gnomAD sites produce comma-separated AF values like `0.25,.` which caused `ValueError` on `float()` conversion during annotation
- Add `_parse_af()` helper that splits on comma and returns the first valid float, handling missing (`.`) values
- Skip records where both parsed AFs are None to avoid unnecessary UPDATEs
- Move progress reporting to top of gnomAD loop so skipped records still advance progress

Fixes #58

## Test plan
- [x] `uv run pytest` — 359 passed, 45 skipped
- [x] `uv run ruff check .` — clean
- [x] New test: multi-allelic AF values (`0.25,.`) parsed correctly
- [x] New test: all-missing AF values (`.,. `) are skipped (count == 0)
- [x] New test: `_parse_af` helper unit tests cover edge cases

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: src/genechat/patch.py [hunk 1]
```diff
@@ -310,6 +310,25 @@ def _update_clinvar_batch(self, batch: list[tuple]) -> int:
...
+        if not raw:
+            return None
+        for part in raw.split(","):
+            part = part.strip()
+            if part and part != ".":
+                try:
+                    return float(part)
+                except ValueError:
+                    continue
+        return None
```
[the remark this answers is not in the document]
Comment (natecostello) [reply src/genechat/patch.py]: Valid concern about allele-index accuracy. However, the current architecture stores a single AF float per annotation row, and the smart filter uses AF > 0.05 as a threshold — for multi-allelic sites, the first non-missing value is a reasonable approximation since allele frequencies at the same position are typically similar in magnitude. A proper fix would require schema changes (storing per-allele AF arrays) and genotype-aware allele selection at query time, which is beyond the scope of this bug fix. The current approach matches how the codebase already handles AF values everywhere else. Not implementing in this PR.

🤖 Generated with [Claude Code](https://claude.ai/code)

File: src/genechat/patch.py [hunk 2]
```diff
@@ -334,10 +353,14 @@ def update_gnomad_from_stream(
                     progress_callback(records_seen)
                     last_report = records_seen
                 continue
+            parsed_af = self._parse_af(af)
+            parsed_af_grpmax = self._parse_af(af_grpmax)
+            if parsed_af is None and parsed_af_grpmax is None:
+                continue
```
[the remark this answers is not in the document]
Comment (natecostello) [reply src/genechat/patch.py]: Applied in commit e557828: moved progress_callback check to top of the loop so all skipped records (both None-from-INFO and None-after-parsing) advance progress reporting.

🤖 Generated with [Claude Code](https://claude.ai/code)

File: src/genechat/patch.py [hunk 3]
```diff
@@ -326,18 +345,22 @@ def update_gnomad_from_stream(
         batch = []
         for record in parse_vcf_stream(stream, ["AF", "AF_grpmax", "AF_popmax"]):
             records_seen += 1
+            if progress_callback and records_seen - last_report >= 10_000:
+                progress_callback(records_seen)
+                last_report = records_seen
             af = record.get("AF")
             # Prefer AF_popmax, fall back to AF_grpmax (gnomAD v4 renamed it)
             af_grpmax = record.get("AF_popmax") or record.get("AF_grpmax")
```
[the remark this answers is not in the document]
Comment (natecostello) [reply src/genechat/patch.py]: This preference order (`AF_popmax` → `AF_grpmax`) is pre-existing logic from the gnomAD v3→v4 migration (v4 renamed AF_popmax to AF_grpmax). It's unrelated to the multi-allelic parsing fix in this PR. The bcftools annotate step only transfers the fields present in the gnomAD VCF, so both won't coexist in practice. Not changing in this PR to keep the fix focused.

🤖 Generated with [Claude Code](https://claude.ai/code)
